### ISSUE TITLE
Occasional "bad map state" during query

### ISSUE THREAD

**Issue by fako1024** [els0r/goProbe#19]
Non-reproducibly, a large query (-30d) on a relatively large DB (~182MB compressed) crashes during query (see below). Based on the paths in the trace the issue occurs with v2.1.1 (built with go 1.11.1).

```root@hh-firewall-1:~# /opt/ntm/goProbe/bin/goQuery -i enp1s0 -n 10  -f -30d agg_talk_port
fatal error: bad map state

goroutine 8 [running]:
runtime.throw(0x5b460b, 0xd)
	/home/fk/Develop/goProbe-2.1.1/go/src/runtime/panic.go:608 +0x72 fp=0xc000473920 sp=0xc0004738f0 pc=0x42bf92
runtime.evacuate(0x57e7e0, 0xc00046e1e0, 0x414)
	/home/fk/Develop/goProbe-2.1.1/go/src/runtime/map.go:1097 +0x657 fp=0xc0004739e0 sp=0xc000473920 pc=0x410927
runtime.growWork(0x57e7e0, 0xc00046e1e0, 0x414)
	/home/fk/Develop/goProbe-2.1.1/go/src/runtime/map.go:1043 +0x61 fp=0xc000473a08 sp=0xc0004739e0 pc=0x410281
runtime.mapassign(0x57e7e0, 0xc00046e1e0, 0xc000473c80, 0xc016485f68)
	/home/fk/Develop/goProbe-2.1.1/go/src/runtime/map.go:579 +0x4be fp=0xc000473a90 sp=0xc000473a08 pc=0x40f3fe
OSAG/goDB.(*DBWorkManager).readBlocksAndEvaluate(0xc000056a40, 0xc000010a00, 0xc00024049b, 0xa, 0xc0002ec000, 0x120, 0x200, 0xc00046e1e0, 0x0, 0x0)
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/goDB/DBWorkManager.go:302 +0xb92 fp=0xc000473ef0 sp=0xc000473a90 pc=0x532be2
OSAG/goDB.(*DBWorkManager).grabAndProcessWorkload(0xc000056a40, 0xc00021cfc0, 0xc00021cf00, 0xc000296360)
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/goDB/DBWorkManager.go:140 +0x127 fp=0xc000473fc0 sp=0xc000473ef0 pc=0x531db7
runtime.goexit()
	/home/fk/Develop/goProbe-2.1.1/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000473fc8 sp=0xc000473fc0 pc=0x457d91
created by OSAG/goDB.(*DBWorkManager).ExecuteWorkerReadJobs
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/goDB/DBWorkManager.go:159 +0xc8

goroutine 1 [semacquire]:
sync.runtime_Semacquire(0xc000296368)
	/home/fk/Develop/goProbe-2.1.1/go/src/runtime/sema.go:56 +0x39
sync.(*WaitGroup).Wait(0xc000296360)
	/home/fk/Develop/goProbe-2.1.1/go/src/sync/waitgroup.go:130 +0x64
OSAG/goDB.(*DBWorkManager).ExecuteWorkerReadJobs(0xc000056a40, 0xc00021cf00)
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/goDB/DBWorkManager.go:168 +0x1de
main.main()
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/query/GPQuery.go:729 +0xae6

goroutine 6 [chan receive]:
main.main.func1(0xc00007eeb0, 0xc0000b0000, 0x414ea08e00000000)
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/query/GPQuery.go:682 +0x76
created by main.main
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/query/GPQuery.go:680 +0x6c4

goroutine 7 [runnable]:
main.aggregate(0xc00021cf00, 0xc00021cf60)
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/query/GPQuery.go:413 +0x3a4
created by main.main
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/query/GPQuery.go:725 +0xa77

goroutine 9 [runnable]:
OSAG/goDB.(*DBWorkManager).readBlocksAndEvaluate(0xc000056a40, 0xc000010a00, 0xc000246a9b, 0xa, 0xc0002dc000, 0x120, 0x200, 0xc00006e5d0, 0x0, 0x0)
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/goDB/DBWorkManager.go:304 +0xc66
OSAG/goDB.(*DBWorkManager).grabAndProcessWorkload(0xc000056a40, 0xc00021cfc0, 0xc00021cf00, 0xc000296360)
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/goDB/DBWorkManager.go:140 +0x127
created by OSAG/goDB.(*DBWorkManager).ExecuteWorkerReadJobs
	/home/fk/Develop/goProbe-2.1.1/addon/gocode/src/OSAG/goDB/DBWorkManager.go:159 +0xc8```

**Comment by els0r** [comment]
Cannot reproduce because it doesn’t show up on the recent develop branch or because the behavior cannot be triggered consistently?

**Comment by fako1024** [comment]
The former - although I run into a different segmentation violation using develop + Go 1.13. The stack is different, but it might actually be the same issue, just the handling might be different in that combination. I'll have to dig into this deeper than expected, this might be some veritable memory management thing. I'll keep you updated.

**Comment by fako1024** [comment]
False alarm... The issue seems to have been caused by a faulty machine - the issue cannot be reproduced on a different machine - plus, by accident, I discovered during debugging that the system was throwing unexpected segmentation violations when executing basic commands on the shell. Digging a little deeper revealed that there seem to be several issues on said machine that cause random segmentation violations. Closing with no-op. Sorry for the wave!
